edx_proctoring/static/proctoring/js/exam_action_handler.js

```diff
@@ -49,11 +49,29 @@ var edx = edx || {};
        });
   };
 
+  function createWorker(url) {
+    var blob, blobBuilder, blobUrl, url;
+    var importScript = "importScripts('" + location.origin + url + "');";
+
+    try {
+      blob = new Blob([importScript], { "type": 'application/javascript' });
+    } catch (e) {
+      //older browsers do not support the Blob constructor, so fallback to the deprecated BlobBuilder
+      //interface in those cases
+      blobBuilder = new (window.BlobBuilder || window.WebKitBlobBuilder || window.MozBlobBuilder)();
+      blobBuilder.append(importScript);
+      blob = blobBuilder.getBlob('application/javascript');
+    }
+
+    url = window.URL || window.webkitURL;
+    blobUrl = url.createObjectURL(blob);
+    return new Worker(blobUrl);
+  }
 
   function workerPromiseForEventNames(eventNames) {
     return function() {
-      var proctoringBackendWorker = new Worker(edx.courseware.proctored_exam.configuredWorkerURL);
-      return new Promise(function(resolve, reject) {
+      var proctoringBackendWorker = createWorker(edx.courseware.proctored_exam.configuredWorkerURL);
+      return new Promise(function (resolve, reject) {
```

Comment:
nit (but would fail eslint if I hadn't dropped the ball on getting it all the way set up):
```suggestion
      return new Promise(function(resolve, reject) {
```